tests/tests_e3/job/scheduling_test.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import absolute_import, division, print_function
+
+import time
+from datetime import timedelta
+
+from e3.collection.dag import DAG
+from e3.job import Job
+from e3.job.scheduler import Scheduler
+
+
+class SleepJob(Job):
+    def run(self):
+        time.sleep(1.0)
+
+
+class TestScheduler(object):
+
+    def test_minimal_run(self):
+        # Test with only two independent jobs
+        dag = DAG()
+        dag.add_vertex('1')
+        dag.add_vertex('2')
+        s = Scheduler(SleepJob, tokens=2)
+        s.run(dag)
+        assert s.stop_time - s.start_time > timedelta(seconds=1)
```

Comment:
I really don’t like such tests that are using sleep(). I would prefer that each job write a file and checking which file is created.